README.md

```diff
@@ -1,2 +1,25 @@
-# pulsar-java-contrib
-Apache pulsar
+# Apache Pulsar Java Contrib
+
+Pulsar java contrib is to provide a non-core code maintenance repository to collect plugin implementations, personalized features, experimental features, and best practices from users.
+
+- [Plugin Contribution Guide](contributionGuides) lists the core interfaces in Pulsar that can be implemented by contributors, and provides implementation guidelines for each type of interface.
+
+- [Plugin Implementation List](contributedFeatures.md) lists the implemented plugins. Users can select the ones they need for reuse.
+
+- [Personalization Features](customizationFeatures.md) lists the customized features and experimental features that require modification to the Pulsar source code.
+
+- [Best Practices]([best-practice-blogs](best-practice-blogs)) lists the best practices for each function summarized by community contributions.
+  - [consume-best-practice.md](best-practice-blogs%2Fconsume-best-practice.md)
+
+This project follows the terms of **Apache License 2.0**.
+You can format the code by `mvn spotless:apply` and generate license headers by `mvn license:format`.
+Please note that the code formatted by Spotless may still not meet the formatting requirements. Please run `mvn checkstyle:check` for inspection.
```

Comment:
This doesn't make much sense. I think it's better to remove checkstyle completely when spotless is used. `spotless:check` should be used to enforce the configured spotless style.
It's necessary to get this right from the beginning, before accepting contributions.